pkg/server/api/bundle/v1/service.go

```diff
@@ -296,7 +296,75 @@ func (s *Service) createFederatedBundle(ctx context.Context, b *types.Bundle, ou
 }
 
 func (s *Service) BatchUpdateFederatedBundle(ctx context.Context, req *bundle.BatchUpdateFederatedBundleRequest) (*bundle.BatchUpdateFederatedBundleResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method BatchUpdateFederatedBundle not implemented")
+	var results []*bundle.BatchUpdateFederatedBundleResponse_Result
+	for _, b := range req.Bundle {
+		results = append(results, s.updateFederatedBundle(ctx, b, req.InputMask, req.OutputMask))
+	}
+
+	return &bundle.BatchUpdateFederatedBundleResponse{
+		Results: results,
+	}, nil
+}
+
+func (s *Service) updateFederatedBundle(ctx context.Context, b *types.Bundle, inputMask, outputMask *types.BundleMask) *bundle.BatchUpdateFederatedBundleResponse_Result {
+	log := rpccontext.Logger(ctx).WithField(telemetry.TrustDomainID, b.TrustDomain)
+
+	td, err := spiffeid.TrustDomainFromString(b.TrustDomain)
+	if err != nil {
+		log.WithError(err).Error("Invalid request: trust domain argument is not valid")
+		return &bundle.BatchUpdateFederatedBundleResponse_Result{
+			Status: api.CreateStatus(codes.InvalidArgument, "trust domain argument is not valid: %q", b.TrustDomain),
```

Comment:
I think you must put error as part of status and not TD, (you can see TD from your resquest)

Reply:
I agree. I put it in that way to be consistent with the other RPCs. However, showing the error looks like a better option. I'll change it in all `*Bundle*` RPCs for consistency.